src/lib/features/NavMenu.svelte

```diff
@@ -1,22 +1,88 @@
 <script lang="ts">
+  import { onMount, tick } from 'svelte';
   import { MenuItem } from '../utils/types.js';
   import Button from '../atoms/Button.svelte';
 
   export let menuItems: MenuItem[] = [];
+  export let highlightMarginTop = '0px';
+
+  let viewportTrackingItems: [string, boolean][] = menuItems
+    .filter((item) => item.viewportHighlightId)
+    .map((item) => [item.viewportHighlightId as string, false]);
+
+  // Order matters, so we just want the
+  $: highlightId = viewportTrackingItems.find(([id, inViewport]) => inViewport === true)?.[0];
+
+  onMount(() => {
```

Comment:
This is the magic that allows the highlight to happen if the `viewportHighlightId` is in the viewport